### PR TITLE
Clean and organize repo before gen3 conversion - tickets/PIPE2D-1527

### PR DESCRIPTION
**Pull request by wtgee**
This PR is a reorganization of the python files and how things are laid out so we can have some consistency going forward and to ease the transition to gen3.

The big change is to split everything into a basic `Task` that does most of the work and a fairly thin `PipelineTask` that handles the `Connections`. This means the `run` method for each given `PipelineTask` is mostly just calling the corresponding `run` method of the `Task`.

Some generic functions have been moved into the `utils` folder, but most logic for a task remains within the task class (or module) itself.

For the gen3 conversion we should only need to update the `PipelineTask` files and hopefully the `Task`s remain mostly the same.

### CODE REVIEW

File: python/pfs/drp/qa/tasks/fiberExtraction.py [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+            "y",
+            "gray",
+            "orange",
+            "cyan",
+            "k",
+        ]
+        return dict(zip(TargetType, itertools.cycle(colors)))
+
+    @classmethod
+    def _makeArgumentParser(cls) -> ArgumentParser:
```
Comment (SogoMineo) [inline python/pfs/drp/qa/tasks/fiberExtraction.py]: Because this class is no longer a subclass of `CmdLineTask`, this method should be removed.

**Comment by SogoMineo** [comment]
The code looks good to me, but I cannot judge whether the policy permits this branch name.

**Comment by wtgee** [comment]
> The code looks good to me, but I cannot judge whether the policy permits this branch name.

Thanks for the review @SogoMineo, I'm just getting back to this as we start back with the gen3 conversion. 

Do you mean the actual branch name (`PIPE2D-1501-01`) or some of the other names within the files? I did convert a few things from camelCase to underscore names, which is in keeping with the lsst dev guide although might not be what we prefer.

**Review by SogoMineo — APPROVED** [review]
Now I have been told that this branch name is not a big deal, I don't think there is a big problem. You should make the commit history clean before you go on to merge.

**Comment by wtgee** [comment]
> Now I have been told that this branch name is not a big deal, I don't think there is a big problem. You should make the commit history clean before you go on to merge.

Will do before the merge, thanks.

Also note that this is not being merged into `main` but into a `gen3` branch which will then be merged into `main` as part of another review.

**Pull request merged**